### PR TITLE
[onnx-3] Make axis on concat required.

### LINKED ISSUE #374: Concat Axis
Unless my protobuf implementation is up the spout (which is possible) the squeezenet model has no axis parameter in the Concat nodes. There is no default indicated. Squeezenet should be concatting on channels. What does no axis parameter indicate since the docs show no default?

### PR DESCRIPTION
**Pull request by ezyang**
Fixes #374

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: onnx/defs/tensor/defs.cc [hunk 1]
```diff
@@ -53,10 +53,11 @@ from the shape argument.)DOC")
 	            "Constrain input and output types to float tensors.");
 
 OPERATOR_SCHEMA(Concat)
+.SinceVersion(3)
 .Attr("axis",
     "Which axis to concat on",
     AttributeProto::INT,
-    OPTIONAL)
+    REQUIRED)
```
Comment (linkerzhang) [inline onnx/defs/tensor/defs.cc]: "required" is the default value, so you may just remove "OPTIONAL" here. Ideally, there're only two APIs, one for required attribute, one for optional attribute with default value (and remove this flag finally).

**Comment by linkerzhang** [comment]
Very interesting. When I tried to change it to "required", it broke several backend test cases (test_backend_test.py, cases: test_densenet121|test_inception_v1|test_inception_v2|test_shufflenet|test_squeezenet). I thought that these test models need to be updated accordingly. It's only my local issue? :)

File: onnx/defs/schema.h [hunk 2]
```diff
@@ -25,6 +25,7 @@ using OperatorSetVersion = int;
 
 const char* const ONNX_DOMAIN = "";
 const bool OPTIONAL = false;
+const bool REQUIRED = true;
```
Comment (linkerzhang) [inline onnx/defs/schema.h]: This may not be needed as the default is "required".

**Comment by houseroad** [comment]
Yeah, I think if we need to regenerate these models. The concats don't have axis in these models.

**Comment by houseroad** [comment]
@anderspapitto could you take a look?

**Comment by anderspapitto** [comment]
I've uploaded a new version of shufflenet, will do the others after I wait a bit to make sure I didn't break anything

**Comment by bddppq** [comment]
@anderspapitto there is no need to “wait”, you can manually test the new model files

**Comment by ezyang** [comment]
good to merge when CI passes

**Pull request merged**